src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import java.io.IOException;
+import java.lang.NullPointerException;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Get and save uploaded image info and return it when requested from front-end. */
+@WebServlet("/get-image-info")
+public class UploadedImageInfoServlet extends HttpServlet {
+  private final BlobstoreService blobstoreService;
+  private String blobKeyString;
```

Comment:
This will lead the competition problem I discussed with you Yesterday. Imagine what will happen if two users (u1, u2) come in the same time:

* u1 does post and set "photoCategory = Product"
* u2 does post and set "photoCategory = ShoppingList"
* u1 does get and will get "photoCategory = ShoppingList" which is wrong.

IIUC, the servlet instance is shared across requests. Thus it can't keep any state about a single request.

Reply:
In the new pr, I used session variable, which should solve this problem.